src/private.rs

```diff
@@ -0,0 +1,118 @@
+use std::fmt;
+
+use super::{AbiEntry, AbiFunction, RootSchema, SCHEMA_VERSION};
+
+pub use schemars;
+use serde::{Deserialize, Serialize};
+
+/// Core ABI information, with schema version and identity hash.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct ChunkedAbiEntry {
+    /// Semver of the ABI schema format.
+    pub(crate) abi_schema_version: String,
+    #[serde(flatten)]
+    pub abi: AbiEntry,
+}
+
+impl ChunkedAbiEntry {
+    pub fn new(functions: Vec<AbiFunction>, root_schema: RootSchema) -> ChunkedAbiEntry {
+        Self {
+            abi_schema_version: SCHEMA_VERSION.to_string(),
+            abi: AbiEntry {
+                functions,
+                root_schema,
+            },
+        }
+    }
+
+    pub fn combine<I: IntoIterator<Item = ChunkedAbiEntry>>(
+        entries: I,
+    ) -> Result<ChunkedAbiEntry, AbiCombineError> {
+        let mut abi_schema_version = None;
+        let mut functions = Vec::<AbiFunction>::new();
+
+        let mut gen = schemars::gen::SchemaGenerator::default();
+        let definitions = gen.definitions_mut();
+
+        let mut unexpected_versions = std::collections::HashSet::new();
```

Comment:
prob better to use a `BTreeSet` since this will be small, and then the errors will be deterministic (small point)